src/go/rpk/pkg/cli/cluster/partitions/move_status.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package partitions
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/docker/go-units"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/types"
+)
+
+func newPartitionMovementsStatusCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var (
+		completion       int
+		all              bool
+		human            bool
+		partitions       []string
+		response         []adminapi.ReconfigurationsResponse
+		filteredResponse []adminapi.ReconfigurationsResponse
+	)
+	cmd := &cobra.Command{
+		Use:   "move-status",
+		Short: "Show ongoing partition movements",
+		Long:  helpListMovement,
+		Run: func(cmd *cobra.Command, topics []string) {
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			// If partition(s) is specified but no topic(s) is specified, exit.
+			if len(topics) <= 0 && len(partitions) > 0 {
+				out.Die("specify at least one topic when --partition is used, exiting.")
+			}
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			response, err = cl.Reconfigurations(cmd.Context())
+			out.MaybeDie(err, "unable to list partition movements: %v\n", err)
+
+			if len(response) == 0 {
+				out.Exit("There are no ongoing partition movements.")
+			}
+
+			for _, t := range topics {
+				nt := strings.Split(t, "/")
+				if len(nt) > 2 {
+					fmt.Printf("invalid format for topic %s, skipping.\n", t)
+					continue
+				}
+				for _, r := range response {
+					isKafkaNs := len(nt) == 1 && r.Ns == "kafka" && r.Topic == t
+					isInternalNs := len(nt) == 2 && r.Ns == nt[0] && r.Topic == nt[1]
+
+					if isKafkaNs || isInternalNs {
+						if len(partitions) == 0 || contains(partitions, strconv.Itoa(r.PartitionID)) {
+							filteredResponse = append(filteredResponse, r)
+						}
+					}
+				}
+			}
+			if len(filteredResponse) > 0 {
+				response = filteredResponse
+			}
+
+			sizeFn := func(size int) string {
+				if human {
+					return units.HumanSize(float64(size))
+				}
+				return strconv.Itoa(size)
+			}
+
+			f := func(rr *adminapi.ReconfigurationsResponse) interface{} {
```

Comment:
I personally think multiple inline functions assigned to variables is tough to read, especially when they aren't being immediately assigned as an input to something else (ex http mux functions to serve endpoints). Any reason to not just have them be standard functions that can be passed in?

Reply:
Oh, this is not part of my change (If you see commit by commit, you'll see that I renamed first the `list.go` -> `move_status.go`, then created a new list.go)

But when you see the whole code change, you'll see like I created move_status.go entirely as part of this PR.

Reply:
Ok reasonable enough. I admit to basically never reviewing commit by commit lol